docs/log_contexts.md

```diff
@@ -385,29 +372,27 @@ def handle_request(request_id):
     return d
 ```
 
-(in general, mixing `with LoggingContext` blocks and
-`defer.inlineCallbacks` in the same function leads to slighly
-counter-intuitive code, under this scheme).
+(in general, mixing `with LoggingContext` blocks and `async` in the same
+function leads to slighly counter-intuitive code, under this scheme).
 
-Because we leave the original `with` block as soon as the Deferred is
+Because we leave the original `with` block as soon as the awaitable is
 returned (as opposed to waiting for it to be resolved, as we do today),
 the logcontext is cleared before control passes back to the reactor; so
 if there is some code within `do_request_handling` which needs to wait
-for a Deferred to complete, there is no need for it to worry about
+for an awaitable to complete, there is no need for it to worry about
 clearing the logcontext before doing so:
 
 ```python
 def handle_request():
-    r = do_some_stuff()
-    r.addCallback(do_some_more_stuff)
-    return r
+    d = do_some_stuff()    # d is a Twisted Deferred
+    d.addCallback(do_some_more_stuff)
+    return d
 ```
 
---- and provided `do_some_stuff` follows the rules of returning a
-Deferred which runs its callbacks in the original logcontext, all is
-happy.
+--- and provided `do_some_stuff` follows the rules of returning an awaitable
+which runs its callbacks in the original logcontext, all is happy.
 
-The business of a Deferred which runs its callbacks in the original
+The business of an awaitable which runs its callbacks in the original
 logcontext isn't hard to achieve --- we have it today, in the shape of
 `context._PreservingContextDeferred`:
```

Comment:
_PreservingContextDeferred was removed back in #2675.

I kinda think this whole section is just a source of confusion rather than providing any useful data now. I don't think the approach it's trying to outline carries over particularly well to `async`/`await`, so I don't think we'd take this approach today. WDYT about just removing it?

Reply:
> _PreservingContextDeferred was removed back in #2675.

Oh, good to know

> WDYT about just removing it?

Yeah agreed, let's do it